dev/integration_tests/flavors/macos/Runner.xcodeproj/project.pbxproj

```diff
@@ -651,6 +651,7 @@
 					"@executable_path/../Frameworks",
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.yourcompany.flavors.free;
+				PRODUCT_FLAVOR = free;
```

Comment:
Why was this needed?  The tool should be parsing the flavor from the configuration, I think?
macOS flavors should work with the same setup as iOS: https://docs.flutter.dev/deployment/flavors#creating-flavors-in-ios

Reply:
It probably isn't needed, but I followed the way that the `iOS` project was set up.

https://github.com/flutter/flutter/blob/997d43618dc73b2d01a86adaccf9dc9672953c03/dev/integration_tests/flavors/ios/Runner.xcodeproj/project.pbxproj#L474

https://github.com/flutter/flutter/blob/997d43618dc73b2d01a86adaccf9dc9672953c03/dev/integration_tests/flavors/ios/Runner/Info-Free.plist#L24

https://github.com/flutter/flutter/blob/997d43618dc73b2d01a86adaccf9dc9672953c03/dev/integration_tests/flavors/ios/Runner/AppDelegate.m#L16-L19

Reply:
Ahh, okay that's the thing that's being used to test that the flavors are hooked up correctly.

So you do need it:
https://github.com/flutter/flutter/pull/117352/files#diff-497597318ad21439d629a57213312c930c1902042dfb2dc31373f0b42fe4d45bR27
https://github.com/flutter/flutter/pull/117352/files#diff-147b89acf8a22478c244cca300415a21dad69ebfdb0ac61051c12b76b1412100R23-R24

Did that test pass before you added this build setting?